remoulade/state/middleware.py

```diff
@@ -20,33 +21,38 @@ def __init__(self, backend, state_ttl=3600, max_size=2e6):
         self.state_ttl = state_ttl
         self.max_size = max_size
 
-    def save(self, message, state):
-        args = message.args
-        kwargs = message.kwargs
+    def save(self, message, state_name, **kwargs):
+        args = str(message.args)
+        kwargs_state = str(message.kwargs)
         message_id = message.message_id
         if sys.getsizeof(args) > self.max_size:
             # Arguments exceed maximum size to display
             #  do not save them.
-            args = []
-        if sys.getsizeof(kwargs) > self.max_size:
+            args = "[]"
+        if sys.getsizeof(kwargs_state) > self.max_size:
             # Keyword arguments exceed maximum size to
             #  display do not save them
-            kwargs = {}
-        self.backend.set_state(State(message_id, state, args, kwargs), self.state_ttl)
+            kwargs_state = "{}"
+        self.backend.set_state(State(message_id, state_name, args, kwargs_state, **kwargs), self.state_ttl)
+
+    def _get_current_time(self):
+        return datetime.utcnow().isoformat()
```

Comment:
As stated in the doc, `datetime.now(timezone.utc)` should be prefered. And call isoformat in State.asdict() : this way you could use the date before formating. And call fromisoformat in from_dict(). The tring is fomatted data, you only want formatted data outside of the application (redis).

https://docs.python.org/fr/3/library/datetime.html#datetime.datetime.utcnow